specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,764 @@
+# App Notification Content Builder
+
+# Background
+
+App Notifications are UI popups that contain rich text, controls and images to display a message to
+the user. With the Windows App SDK, developers can define an XML payload and display an
+AppNotification using the AppNotificationManager::Show API, but defining these XML payloads can be
+tedious when written by hand.
+
+To generate an AppNotification like this one:
+
+![App Notification With Avatar Image](AppNotificationWithAvatarImage.png)
+
+Developers have to write something like this:
+
+```c++
+int sequenceId {1234};
+std::wstring pathToImage {LR"(path\to\my\image.png)"};
+
+winrt::hstring xmlPayload{
+    L"<toast launch = \"action=AppNotificationClick&amp;sequence=" + to_wstring(sequenceId) + L"\">\
+        <visual>\
+            <binding template = \"ToastGeneric\">\
+                <image placement = \"appLogoOverride\" hint-crop=\"circle\" src = \"" + pathToImage + L"\"/>\
+                <text>App Notification with Avatar Image</text>\
+                <text>This is an example message using XML</text>\
+            </binding>\
+        </visual>\
+        <actions>\
+            <action\
+                content = \"Open App\"\
+                arguments = \"action=OpenApp&amp;sequence=" + to_wstring(sequenceId) + L"\"/>\
+        </actions>\
+    </toast>" };
+
+auto appNotification{ winrt::AppNotification(xmlPayload) };
+winrt::AppNotificationManager::Default().Show(appNotication);
+```
+
+# Description
+
+To improve the AppNotification construction experience, the Windows App SDK is introducing
+AppNotificationContent, a builder that constructs simple or rich App Notifications for the
+developer.
+
+```c++
+AppNotificationContent()
+    .AddArgument(L"AppNotificationClick")
+    .AddArgument(L"sequence", 1234)
+    .AddImage(Image(LR"(path\to\my\image.png)")
+    .SetImagePlacement(ImagePlacement::AppLogoOverride)
+        .SetUsesCircleCrop(true))
```

Comment:
.SetUsesCircleCrop(true)) is too verbose. Can we explore a less verbose version? Another idea: All AppLogoOverride images are cropped to circles by default, and developers who do not want the images cropped can set a flag for it to not be cropped.

Reply:
Ignore my idea, inline images can also be cropped.